lib/WUI/nhttp/gcode_upload.h

```diff
@@ -46,14 +46,22 @@ class GcodeUpload final : private UploadHooks {
     virtual Result finish(const char *final_filename, bool start_print) override;
     virtual Result check_filename(const char *filename) const override;
 
+    // filepath in case of PUT (we know it from the start)
+    char filepath [FILE_PATH_BUFFER_LEN];
```

Comment:
The filepath is somewhat big thing. As we have either the uploader or the filepath, we could put these into a std::variant to save valuable space.

Then we'll also be able to dispatch between them by `std::visit` and an overloaded method instead of `stepPOST` and `stepPUT`.